### PR TITLE
Update upgrade path so Akismet Free to Akismet Personal is set to

### PR DESCRIPTION
**Pull request by CodeyGuyDylan**
## Proposed Changes

This PR updates the Akismet upgrade paths so that when a user upgrades from Akismet Free, the Akismet Personal plan (aka the Pay What You Want plan) defaults to the average $36

NOTE: I also used this opportunity to improve the Akismet products, fix a typo, fix a bug, and add some tests for the upgrade button

## Testing Instructions

1. Checkout this branch & Start your local environment
2. Sandbox `public-api.wordpress.com`
3. **Make sure you are using the store sandbox**
4. Make sure you have no Akismet plans active on your account ( you can delete them from the Store Admin )
5. Go to http://calypso.localhost:3000/checkout/akismet/ak_free_yearly and complete the checkout
6. Head to http://calypso.localhost:3000/me/purchases and select your free Akismet plan
7. Click "Upgrade"
8. Make sure that the PWYW plan defaults to $36
image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/calypso-products/src/constants/akismet.ts [hunk 1]
```diff
@@ -18,71 +18,86 @@ export const PRODUCT_AKISMET_ENTERPRISE_2M_MONTHLY = 'ak_ep2m_monthly_1';
...
 
+export const AKISMET_BUSINESS_MONTHLY_PRODUCTS = <const>[ PRODUCT_AKISMET_ENTERPRISE_MONTHLY ];
+
 export const AKISMET_ENTERPRISE_MONTHLY_PRODUCTS = <const>[
-	PRODUCT_AKISMET_ENTERPRISE_2M_YEARLY,
+	PRODUCT_AKISMET_ENTERPRISE_350K_MONTHLY,
 	PRODUCT_AKISMET_ENTERPRISE_2M_MONTHLY,
-	PRODUCT_AKISMET_ENTERPRISE_GT2M_YEARLY,
 	PRODUCT_AKISMET_ENTERPRISE_GT2M_MONTHLY,
 ];
```
Comment (CodeyGuyDylan) [inline packages/calypso-products/src/constants/akismet.ts:36]: This previously had a couple YEARLY products in it, and it was specifically meant for MONTHLY products, went ahead and fixed this up

File: packages/calypso-products/src/constants/akismet.ts [hunk 2]
```diff
@@ -18,71 +18,86 @@ export const PRODUCT_AKISMET_ENTERPRISE_2M_MONTHLY = 'ak_ep2m_monthly_1';
...
 export const AKISMET_PLUS_PRODUCTS = <const>[
 	...AKISTMET_PLUS_MONTHLY_PRODUCTS,
 	...AKISMET_PLUS_YEARLY_PRODUCTS,
 ];
 
+export const AKISMET_BUSINESS_PRODUCTS = <const>[
+	...AKISMET_BUSINESS_MONTHLY_PRODUCTS,
+	...AKISMET_BUSINESS_YEARLY_PRODUCTS,
+];
+
```
Comment (CodeyGuyDylan) [inline packages/calypso-products/src/constants/akismet.ts:69]: Went ahead and extracted the business products from enterprise as well

File: packages/calypso-products/src/constants/akismet.ts [hunk 3]
```diff
@@ -95,4 +110,8 @@ export const AKISMET_UPGADES_PRODUCTS_MAP = {
 	[ PRODUCT_AKISMET_PLUS_40K_YEARLY ]: `/checkout/akismet/${ PRODUCT_AKISMET_ENTERPRISE_YEARLY }`,
 	[ PRODUCT_AKISMET_ENTERPRISE_MONTHLY ]: 'https://akismet.com/enterprise',
 	[ PRODUCT_AKISMET_ENTERPRISE_YEARLY ]: 'https://akismet.com/enterprise',
+	[ PRODUCT_AKISMET_ENTERPRISE_350K_MONTHLY ]: 'https://akismet.com/enterprise',
+	[ PRODUCT_AKISMET_ENTERPRISE_350K_YEARLY ]: 'https://akismet.com/enterprise',
+	[ PRODUCT_AKISMET_ENTERPRISE_2M_MONTHLY ]: 'https://akismet.com/enterprise',
+	[ PRODUCT_AKISMET_ENTERPRISE_2M_YEARLY ]: 'https://akismet.com/enterprise',
```
Comment (CodeyGuyDylan) [inline packages/calypso-products/src/constants/akismet.ts:116]: I figured any plan that can possibly be upgraded should still have an upgrade button, so I added these enterprise plans with an upgrade button that will go to `/enterprise` excluding the top enterprise plans ( GT2M yearly/monthly )

File: packages/calypso-products/src/constants/akismet.ts [hunk 4]
```diff
@@ -18,71 +18,86 @@ export const PRODUCT_AKISMET_ENTERPRISE_2M_MONTHLY = 'ak_ep2m_monthly_1';
...
 	...AKISMET_PERSONAL_PRODUCTS,
 	...AKISMET_PLUS_PRODUCTS,
+	...AKISMET_BUSINESS_PRODUCTS,
 	...AKISMET_ENTERPRISE_PRODUCTS,
 ];
 
-export const AKISMET_UPGADES_PRODUCTS_MAP = {
-	[ PRODUCT_AKISMET_FREE ]: `/checkout/akismet/${ PRODUCT_AKISMET_PERSONAL_YEARLY }`,
+export const AKISMET_UPGRADES_PRODUCTS_MAP = {
+	[ PRODUCT_AKISMET_FREE ]: `/checkout/akismet/${ PRODUCT_AKISMET_PERSONAL_YEARLY }:-q-36`,
```
Comment (CodeyGuyDylan) [inline packages/calypso-products/src/constants/akismet.ts:99]: This was the main reason for this diff, defaulting the Personal upgrade to $36

File: client/me/purchases/manage-purchase/index.jsx [hunk 5]
```diff
@@ -32,7 +32,7 @@ import {
 	isDIFMProduct,
 	isAkismetProduct,
 	isJetpackBackupT1Slug,
-	AKISMET_UPGADES_PRODUCTS_MAP,
+	AKISMET_UPGRADES_PRODUCTS_MAP,
```
Comment (CodeyGuyDylan) [inline client/me/purchases/manage-purchase/index.jsx:35]: Fixed typo here

Comment (andrii-lysenko) [reply client/me/purchases/manage-purchase/index.jsx:35]: Thanks for spotting this.

**Review by andrii-lysenko — APPROVED** [review]
The changes looks good and work as described. 

The tests are also a great addition 👍

File: client/me/purchases/manage-purchase/index.jsx (hunk 5 above)
Comment (IanRamosC) [reply client/me/purchases/manage-purchase/index.jsx:35]: Yeah, nice catch. 🎉

**Review by IanRamosC — APPROVED** [review]
Tests well! :shipit:

**Pull request merged**